modules/auxiliary/scanner/dcerpc/petitpotam.rb

```diff
@@ -66,8 +66,17 @@ def initialize
   end
 
   def run_host(_ip)
-    connect
-    smb_login
+    begin
+      connect
+    rescue Rex::ConnectionError => e
+      fail_with(Failure::Unreachable, e.message)
+    end
+
+    begin
+      smb_login
+    rescue Rex::Proto::SMB::Exceptions::Error, RubySMB::Error::RubySMBError => e
+      fail_with(Failure::NoAccess, "Unable to authenticate ([#{e.class}] #{e}).")
+    end
```

Comment:
Forgot to mention that I added this error handling while testing this module to ensure they both behave in the same way. It was throwing exceptions and displaying stack traces because I misconfigured the credentials. This prevents that and should make it clear that the issue is that it's unable to authenticate.